launcher/ui/pages/global/AccountListPage.cpp

```diff
@@ -134,8 +135,19 @@ void AccountListPage::listChanged()
 
 void AccountListPage::on_actionAddMicrosoft_triggered()
 {
-    MinecraftAccountPtr account =
-        MSALoginDialog::newAccount(this, tr("Please enter your Mojang account email and password to add your account."));
+    QMessageBox box(this);
+    box.setWindowTitle(tr("Add account"));
+    box.setText(tr("How do you want to login?"));
+    box.setIcon(QMessageBox::Question);
+    auto deviceCode = box.addButton(tr("Legacy"), QMessageBox::ButtonRole::YesRole);
+    auto authCode = box.addButton(tr("Recommended"), QMessageBox::ButtonRole::NoRole);
+    auto cancel = box.addButton(tr("Cancel"), QMessageBox::ButtonRole::RejectRole);
+    box.setDefaultButton(cancel);
```

Comment:
shouldn't the default button be "Reccomended"?

Reply:
I put cancel as the default because the default button is also considered clicked, and if I exit the message box by closing it(not hitting any button) I do not want to proceed with the login.

Reply:
it's still kinda weird to default to cancel, cuz I can see people may just want to just press enter and login, without having to move to recommended

Reply:
can't you handle the close button manually? Or does the message box automatically do the reject/cancel action when closed? That would be the expected behaviour to me... 
I can't see anything obvious in the docs, but haven't worked with it so far either

Reply:
Sorry, I was wrong it looks like you are right. I made the recommended button default.